src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs

```diff
@@ -123,12 +126,189 @@ public static IResourceBuilder<RedisResource> WithRedisCommander(this IResourceB
+                        writer.WriteNumber("port", endpoint.TargetPort!.Value);
+                        writer.WriteString("name", redisResource.Name);
+                        writer.WriteNumber("db", 0);
+                        //todo: provide username and password when https://github.com/dotnet/aspire/pull/4642 merged.
+                        writer.WriteNull("username");
+                        writer.WriteNull("password");
+                        writer.WriteString("connectionType", "STANDALONE");
+                        writer.WriteEndObject();
+                    }
+                }
+                writer.WriteEndArray();
+                await writer.FlushAsync(ct).ConfigureAwait(false);
+                stream.Seek(0, SeekOrigin.Begin);
+
+                var content = new MultipartFormDataContent();
+
+                var fileContent = new StreamContent(stream);
+
+                content.Add(fileContent, "file", "RedisInsight_connections.json");
+
+                var apiUrl = $"/api/databases/import";
+
+                var pipeline = new ResiliencePipelineBuilder().AddRetry(new Polly.Retry.RetryStrategyOptions
+                {
+                    Delay = TimeSpan.FromSeconds(2),
+                    MaxRetryAttempts = 5,
+                }).Build();
+
+                try
+                {
+                    await pipeline.ExecuteAsync(async (ctx) =>
+                    {
+                        var response = await client.PostAsync(apiUrl, content, ctx)
+                        .ConfigureAwait(false);
+
+                        response.EnsureSuccessStatusCode();
+                    }, ct).ConfigureAwait(false);
+
+                }
+                catch (Exception ex)
+                {
+                    resourceLogger.LogError("Could not import Redis databases into RedisInsight. Reason: {reason}", ex.Message);
+                }
+            };
+        }
+
+        static async Task AcceptRedisInsightEula(ILogger resourceLogger, HttpClient client, CancellationToken ct)
+        {
+            using (var stream = new MemoryStream())
+            {
+                using var writer = new Utf8JsonWriter(stream);
+
+                writer.WriteStartObject();
+
+                writer.WritePropertyName("agreements");
+                writer.WriteStartObject();
+                writer.WriteBoolean("eula", true);
+                writer.WriteBoolean("analytics", false);
+                writer.WriteBoolean("notifications", false);
+                writer.WriteBoolean("encryption", false);
+                writer.WriteEndObject();
+
+                writer.WriteEndObject();
+
+                await writer.FlushAsync(ct).ConfigureAwait(false);
+                stream.Seek(0, SeekOrigin.Begin);
+                string json = Encoding.UTF8.GetString(stream.ToArray());
+                var content = new StringContent(json, Encoding.UTF8, "application/json");
```

Comment:
If you want to write a follow up @Alirexaa, this can be simplified using JsonContent and JsonObject.